src/cudadecoder/online-endpoint.cc

```diff
@@ -0,0 +1,71 @@
+// online2/online-endpoint.cc
+
+// Copyright    2014  Johns Hopkins University (author: Daniel Povey)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cudadecoder/online-endpoint.h"
+
+namespace kaldi {
```

Comment:
I do not think we can get away with duplicate symbols for struct and function names. The simplest way is to put this simplified version into the same `kaldi::cuda_decoder` namespace as everything else here.

Can you think of a way to reuse code at least partially? E.g., split the original file into online-enpoint-lite and online-endpoint? At the very least, add a comment that this is a tear-off  copypasta of `online2/online-endpoint.h`. But of we can avoid cloning 500 lines, it would be much better.